rust/lance/src/dataset.rs

```diff
@@ -1333,11 +1241,123 @@ impl Dataset {
     ///
     /// This is a metadata-only operation and does not remove the data from the
     /// underlying storage. In order to remove the data, you must subsequently
-    /// call `compact_files` to rewrite the data without the removed columns and
-    /// then call `cleanup_old_versions` to remove the old files.
+    /// call [optimize::compact_files()] to rewrite the data without the removed columns and
+    /// then call [cleanup::cleanup_old_versions()] to remove the old files.
     pub async fn drop_columns(&mut self, columns: &[&str]) -> Result<()> {
         schema_evolution::drop_columns(self, columns).await
     }
+
+    /// Drop columns from the dataset and return updated dataset. Note that this
+    /// is a zero-copy operation and column is not physically removed from the
+    /// dataset.
+    /// Parameters:
+    /// - `columns`: the list of column names to drop.
+    #[deprecated(since = "0.9.12", note = "Please use `drop_columns` instead.")]
+    pub async fn drop(&mut self, columns: &[&str]) -> Result<()> {
+        self.drop_columns(columns).await
+    }
```

Comment:
This has been deprecated a while, can we just remove it?

Reply:
I can do that in a follow up. Will better show up in change notes if I do a dedicated PR.